In [1]:
real_estate = read.csv("real_estate_p.csv")

In [2]:
head(real_estate)

Price  Lot.Size Waterfront Age Land.Value New.Construct Central.Air
1 132500 0.09     0           42 50000      0             0          
2  90000 0.83     0           36 22200      0             0          
3 122900 1.94     0            4 21200      0             0          
4 325000 2.29     0          123 12600      0             0          
5 120000 0.92     0            1 22300      0             0          
6  87500 0.88     0           17 19400      0             0          
  Living.Area Bedrooms Fireplaces Bathrooms Rooms
1  906        2        1          1.0        5   
2 1632        3        0          1.5        6   
3 1416        3        0          1.5        6   
4 2894        7        0          1.0       12   
5 1624        3        0          2.0        6   
6 1092        3        0          1.0        6

Real estate agents know the three most important factors in determining the price of a
house are location, location, and location. But what other factors help determine the price
at which a house should be listed? Number of bathrooms? Size of the yard?
A random sample is drawn from publicly available data on thousands of homes in upstate
New York. The sample data are saved in the file ‘real_estate_p.csv’. The variables include the
price (in dollars), total living area (in square feet), number of bathrooms, number of
bedrooms, size of lot (in acres), age of house (in years), etc.
The part B of the project aims at building a linear model to predict the selling price using a
set of potential predictor variables. Analysis of nontrivial data entails lengthy calculations,
which should be performed by utilizing statistical software. You are required to manage the
data set and to obtain important results from data analysis through effective use of R. At
the same time, you may appreciate the role of linear regression models in scientific inquiry. 

Model #1: a multiple regression model including the living area ($X_{1}$), number of bedrooms
($X_{2}$), and number of fireplaces ($X_{3}$) as predictor variables.
<br/> 
### (1) State the fitted regression equation. 

In [3]:
multireg = lm(Price~Living.Area+Bedrooms+Fireplaces, data = real_estate)
summary(multireg)


Call:
lm(formula = Price ~ Living.Area + Bedrooms + Fireplaces, data = real_estate)

Residuals:
    Min      1Q  Median      3Q     Max 
-200323  -38243   -5742   28241  306139 

Coefficients:
              Estimate Std. Error t value Pr(>|t|)    
(Intercept)  28357.351  14661.995   1.934   0.0542 .  
Living.Area    131.440      9.405  13.975   <2e-16 ***
Bedrooms    -15873.874   6355.692  -2.498   0.0131 *  
Fireplaces    1138.275   7961.437   0.143   0.8864    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 61070 on 259 degrees of freedom
Multiple R-squared:  0.5872,	Adjusted R-squared:  0.5824 
F-statistic: 122.8 on 3 and 259 DF,  p-value: < 2.2e-16


Price = 28357.351 + 131.44Living.Area - 15873.874Bedrooms + 1138.275Fireplaces

### (2) Interpret the coefficient of the number of bedrooms ($X_{2}$). 

When all other predictors are held constant increasing the number of bedrooms by 1 will decrease the price by $15873.87.

### (3) Produce an ANOVA table. Report SST, SSR, and SSE, and their corresponding degrees of freedom. 

In [6]:
anv = anova(multireg)
anv

Df  Sum Sq       Mean Sq      F value      Pr(>F)      
Living.Area   1 1.350146e+12 1.350146e+12 362.03939361 4.209785e-51
Bedrooms      1 2.364153e+10 2.364153e+10   6.33943642 1.241203e-02
Fireplaces    1 7.623187e+07 7.623187e+07   0.02044145 8.864222e-01
Residuals   259 9.658833e+11 3.729279e+09           NA           NA

In [14]:
ssr = sum(anv[,2])-anv[4,2]
ssr

[1] 1.373864e+12

In [15]:
sse = anv[4,2]
sse

[1] 965883347601

In [16]:
sst = ssr + sse
sst

[1] 2.339747e+12

### 4) Using $\alpha$ = 0.05, perform an F test of overall linear relationship. State the hypotheses, the value of F-test statistic, p-value, and your conclusion. 